flow/view_slicer.h

```diff
@@ -0,0 +1,25 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_FLOW_VIEW_SLICER_H_
+#define FLUTTER_FLOW_VIEW_SLICER_H_
+
+#include <unordered_map>
+#include "display_list/dl_canvas.h"
+#include "flow/embedded_views.h"
+
+namespace flutter {
+
+/// @brief Compute the required overlay layers and clip the view slices
```

Comment:
Can this documentation be elaborated? What are the integer handles? How are they created. What is the rect that is returned? The embedder view slice is also not documented so there are no breadcrumbs there either.

Can these go in a struct with a custom hasher and equality checker?

If this is just moving code around and you want to limit scope, then we could probably do it in a followup. Just that from the perspective of someone not super familiar with slicing, this reads a bit dense.

Reply:
I agree for the most part, the platform view code is not very abstracted - lots of vecs/maps with the integer id keys that should be refactored. I'd like to do that sort of gradually though, since its very related to the work in

https://github.com/flutter/engine/pull/53882 and https://github.com/flutter/engine/pull/53826

Reply:
sgtm